app/services/paymentsAndCharges/PaymentsAndChargesService.scala

```diff
@@ -18,89 +18,103 @@ package services.paymentsAndCharges
 
 import java.time.LocalDate
 
+import connectors.cache.FinancialInfoCacheConnector
 import controllers.chargeB.{routes => _}
-import helpers.FormatHelper
+import helpers.FormatHelper._
+import javax.inject.Inject
 import models.LocalDateBinder._
 import models.financialStatement.SchemeFS
 import models.financialStatement.SchemeFSChargeType.{PSS_AFT_RETURN, PSS_AFT_RETURN_INTEREST, PSS_OTC_AFT_RETURN, PSS_OTC_AFT_RETURN_INTEREST}
 import models.viewModels.paymentsAndCharges.PaymentAndChargeStatus.{InterestIsAccruing, NoStatus, PaymentOverdue}
 import models.viewModels.paymentsAndCharges.{PaymentsAndChargesDetails, PaymentsAndChargesTable}
 import play.api.i18n.Messages
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.viewmodels.SummaryList.{Key, Row, Value}
 import uk.gov.hmrc.viewmodels.Text.Literal
 import uk.gov.hmrc.viewmodels.{Content, Html, SummaryList, _}
 import utils.DateHelper.{dateFormatterDMY, dateFormatterStartDate}
 import viewmodels.Table
 import viewmodels.Table.Cell
 
-class PaymentsAndChargesService {
-
-  def getPaymentsAndChargesSeqOfTables(paymentsAndChargesForAGivenPeriod: Seq[(LocalDate, Seq[SchemeFS])], srn: String)(
-      implicit messages: Messages): Seq[PaymentsAndChargesTable] = {
-
-    paymentsAndChargesForAGivenPeriod.map { paymentsAndCharges =>
-      val seqPaymentsAndCharges = paymentsAndCharges._2
-
-      val seqPayments = seqPaymentsAndCharges.flatMap { details =>
-        val onlyAFTAndOTCChargeTypes = details.chargeType == PSS_AFT_RETURN || details.chargeType == PSS_OTC_AFT_RETURN
-
-        val redirectChargeDetailsUrl = controllers.paymentsAndCharges.routes.PaymentsAndChargeDetailsController
-          .onPageLoad(srn, details.periodStartDate, details.chargeReference)
-          .url
-
-        (onlyAFTAndOTCChargeTypes, details.amountDue > 0) match {
-
-          case (true, true) if details.accruedInterestTotal > 0 =>
-            createPaymentAndChargesIfInterestAccrued(details, srn)
-          case (true, _) if details.totalAmount < 0 => Seq.empty
-          case _ =>
-            Seq(
-              PaymentsAndChargesDetails(
-                details.chargeType.toString,
-                details.chargeReference,
-                s"${FormatHelper.formatCurrencyAmountAsString(details.amountDue)}",
-                NoStatus,
-                redirectChargeDetailsUrl,
-                messages(s"paymentsAndCharges.visuallyHiddenText", details.chargeReference)
-              ))
+import scala.concurrent.{ExecutionContext, Future}
+
+class PaymentsAndChargesService @Inject()(fiCacheConnector: FinancialInfoCacheConnector) {
+
+  def getPaymentsAndCharges(paymentsAndChargesForAGivenPeriod: Seq[(LocalDate, Seq[SchemeFS])], srn: String)
+                           (implicit messages: Messages, ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[PaymentsAndChargesTable]] = {
+
+    fiCacheConnector.fetch map {
```

Comment:
This method needs breaking up - it's too long

Reply:
Done